include/linker/common-ram.ld

```diff
@@ -14,18 +14,27 @@
 #endif
 
 /*
- * Space for storing per device busy bitmap. Since we do not know beforehand
- * the number of devices, we go through the below mechanism to allocate the
- * required space.
+ * Space for storing per device init status and busy bitmap in case PM is
+ * enabled. Since we do not know beforehand the number of devices,
+ * we go through the below mechanism to allocate the required space.
+ * Both are made of 1 bit per-device instance, so we compute the size of
+ * of an entire bitfield, aligned on 32bits.
  */
-#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
 #define DEVICE_COUNT \
 	((__device_end - __device_start) / _DEVICE_STRUCT_SIZEOF)
-#define DEV_BUSY_SZ	(((DEVICE_COUNT + 31) / 32) * 4)
+#define DEVICE_BITFIELD_SIZE	(((DEVICE_COUNT + 31) / 32) * 4)
+
+#define DEVICE_INIT_STATUS_BITFIELD() 		\
+		FILL(0xFF);			\
```

Comment:
I followed the legacy behavior: driver_api is set at build time, and unset if init fails. Could be other way round (though it means it will set the bit for each and every device, where here it unsets if only init fails. probably not a performance hit though)

Reply:
I'll switch to the other way round, which makes more sense.